### DISCUSSION TITLE
#magick' error messages

### DISCUSSION THREAD

**Discussion by claus-01** [ImageMagick/ImageMagick#2321] [Digital Image Processing]
I'm trying to get a shell script to run that I made myself using one of the [IM examples](http://www.imagemagick.org/Usage/canvas/#sparse_fill), but I am getting error messages:

```
`convert-im6.q16: attempt to perform an operation not allowed by the security policy `@-' @ error/property.c/InterpretImageProperties/3666.`
`convert-im6.q16: memory allocation failed SparseColorOption @ error/mogrify.c/SparseColorOption/535.`
`convert-im6.q16: attempt to perform an operation not allowed by the security policy `@-' @ error/property.c/InterpretImageProperties/3666.`
```

And the script:

```
#!bin/sh

magick sparse_source.gif txt:- |\
    sed '1d; / 0) /d; s/:.* /,/;' |\
      magick sparse_source.gif -alpha off \
              -sparse-color shepards '@-' sparse_fill.png
```

Does anyone know what they mean? I have already posted in the Ubuntu Forum, but got no answer. And yes, I have sed installed.

TIA,

Claus

P. S.: I erroneously first used 'convert' instead of 'magick'. Now there are no error messages anymore, but the script seems to run indefinitely. I'm not getting the system prompt again.

**Comment by urban-warrior** [comment]
Ubuntu has a strict security policy.  You will need to remove the policy that prevents indirect files from being read specified with the '@' character.  Locate the policy.xml file and comment out that restrictive policy and try again.  See https://imagemagick.org/script/security-policy.php for more information about the ImageMagick security policy.  Alternatively, don't use the '@' character.  Instead, try this command:

```
magick sparse_source.gif txt:- |\
    sed '1d; / 0) /d; s/:.* /,/;' > sparse.txt
magick sparse_source.gif -alpha off \
    -sparse-color shepards sparse.txt sparse_fill.png
```

**Reply by claus-01** [reply]
Thanks.

Claus

**Comment by claus-01** [comment]
Hi,

`magick sparse_source.gif -alpha off sparse-alpha.png \ magick sparse-alpha.png  -sparse-color shepards '@-' sparse_fill.png`

results in

```
magick: unable to open image ' magick': No such file or directory @ error/blob.c/OpenBlob/3539.
magick: no decode delegate for this image format `' @ error/constitute.c/ReadImage/562.

```

What am I doing wrong here? I copied your code straight into the shell.

Claus

**Reply by fmw42** [reply]
You need to edit your policy.xml file to give read|write permission for @ Also \ means you are going to have  a new line. You show only one line of command. Your @- symbolizes that you are getting coordinates from stdout/stdin. But I see no stdout to pass the coordinates to the @-.  See https://imagemagick.org/Usage/canvas/#sparse_summery. I think you may be confusing -sparse-color with sparse-color:. See that under formats at https://imagemagick.org/script/formats.php

**Reply by claus-01** [reply]
Hello Fred,

unfortunately, I don't have a 'policy.xml' on my computer. I only have the 'magick' binary. (I searched my entire $HOME, but couldn't find anything. I'm using Ubuntu 20.04.)

Claus 

P. S.: After reading part of IM's policy, I typed 'magick identify -list policy' in a shell and got

`Path: [built-in]
  Policy: Undefined
    rights: None`

May I write my own 'policy.xml'? Where do I save it? My 'magick' binary is in '$HOME/Software/ImageMagick'.

**Reply by urban-warrior** [reply]
Copy/paste error on our part.  We correct the command in the original post.

**Comment by claus-01** [comment]
Shouldn't line 2 `sed ...` end with `\`, like

```
magick sparse_source.gif txt:- |\
    sed '1d; / 0) /d; s/:.* /,/;' > sparse.txt \
magick sparse_source.gif -alpha off \
    -sparse-color shepards sparse.txt sparse_fill.png

```
Claus
